app/controllers/automated_tests_controller.rb

```diff
@@ -114,43 +96,17 @@ def execute_test_run
     redirect_to action: :student_interface, id: params[:id]
   end
 
-  # Download is called when an admin wants to download a test script
-  # or test support file
-  # Check three things:
-  #  1. filename is in DB
-  #  2. file is in the directory it's supposed to be
-  #  3. file exists and is readable
+  # Download is called when an admin wants to download a test file
   def download
-    if params[:type] == 'script'
-      model_class = TestScript
-    else # params[:type] == 'support'
-      model_class = TestSupportFile
-    end
-    filedb = model_class.find_by(assignment_id: params[:assignment_id], file_name: params[:filename])
-
-    if filedb
-      filename = filedb.file_name
-      assn_short_id = Assignment.find(params[:assignment_id]).short_identifier
-
-      # the given file should be in this directory
-      should_be_in = File.join(AutomatedTestsClientHelper::ASSIGNMENTS_DIR, assn_short_id)
-      should_be_in = File.expand_path(should_be_in)
-      filename = File.expand_path(File.join(should_be_in, filename))
-
-      if should_be_in == File.dirname(filename) and File.readable?(filename)
-        # Everything looks OK. Send the file over to the client.
-        file_contents = IO.read(filename)
-        send_file filename,
-                  type: ( SubmissionFile.is_binary?(file_contents) ? 'application/octet-stream':'text/plain' ),
-                  x_sendfile: true
-
-        # print flash error messages
-      else
-        flash_message(:error, I18n.t('automated_tests.download_wrong_place_or_unreadable'))
-        redirect_to action: 'manage'
-      end
+    assignment = Assignment.find(params[:assignment_id])
+    file_path = File.join(AutomatedTestsClientHelper::ASSIGNMENTS_DIR, assignment.short_identifier, params[:filename])
+    if File.exist?(file_path)
+      file_contents = IO.read(file_path)
+      send_file file_path,
+                type: ( SubmissionFile.is_binary?(file_contents) ? 'application/octet-stream':'text/plain' ),
```

Comment:
Not sure why this isn't a Rubocop thing, but I'd prefer having spaces around the `:` in the ternary if.